### ISSUE TITLE
Schema import deadlock situation

### ISSUE THREAD

**Issue by etiennedi** [creativesoftwarefdn/weaviate#851]
Possibly related to #811, but this time not for updating, but for creating classes.

crash looping with 
```
weaviate_1             | {"action":"contextionary_rebuild","error":"Could not build in-memory contextionary from schema; Could not find camel cased part of name 'warld' for property warld in class 'Geopolymer' in the contextionary, consider adding some keywords instead.","level":"fatal","msg":"could not (re)build contextionary","time":"2019-04-29T12:12:17Z"}
```

It seemed to have passed the validation on adding fine, but then failed on rebuilding the c11y.

If the user gets stuck in this situation they cannot recover from it.

**Comment by bobvanluijt** [comment]
I think that the flakiness of the current C11y also plays a role here.

**Comment by etiennedi** [comment]
I don't disagree, but nevertheless weaviate must be resilient enough that you cannot break the entire setup due to bad user input (whether it's due to the c11y or not). I think I found the root cause, fix coming up soon.

**Comment by etiennedi** [comment]
**Root cause analysis**:

There is test logic in the non-test code of the scheme manager. There is a check if the c11y is set to nil, we ignore all checks on it. From what I can tell this was done to avoid writing an actual fake for the tests, so that the tests would just skip it. However, this also led to an error where we passed a nil-contextionary in production. Without the "unit test workaround" this would have been easily caught in our acceptance tests, but this way it didn't. 

I'll remove the unit test workaround, to make sure we actually test the same code we run in production. In the mean time I have already fixed the issue of why the c11y was nil in the first place. This fix is coming up as soon as my local test run has passed.
